### PR TITLE
Add TF2 compatibility changes that are low-hanging fruit

### PR DESCRIPTION
**Pull request by danieldk**

### CODE REVIEW

File: sticker-graph/sticker_graph/model.py [hunk 1]
```diff
@@ -24,21 +24,21 @@ def affine_transform(self, prefix, x, n_outputs):
...
         return tf.reshape(
-            tf.nn.xw_plus_b(
+            tf.compat.v1.nn.xw_plus_b(
                 x, w, b), [
                 batch_size, -1, n_outputs])
 
     def decode_word(self, word, max_subword_len):
-        bytes = tf.decode_raw(word, tf.uint8)[:max_subword_len]
-        return tf.pad(bytes, [[0, max_subword_len - tf.shape(bytes)[0]]])
+        bytes = tf.io.decode_raw(word, tf.uint8)[:max_subword_len]
```
Comment (sebpuetz) [inline sticker-graph/sticker_graph/model.py:40]: Unrelated, but isn't `bytes` a built-in?

**Review by sebpuetz — APPROVED** [review]
Didn't see anything odd, but maybe @twuebi is deeper into the matter.

File: sticker-graph/sticker_graph/model.py (hunk 1 above)
Comment (danieldk) [reply sticker-graph/sticker_graph/model.py:40]: Not a keyword:
https://docs.python.org/3.7/reference/lexical_analysis.html#keywords

Which is probably why this is permitted. But it is indeed a built-in class, so this should probably be renamed.

File: sticker-graph/sticker_graph/write_helper.py [hunk 2]
```diff
@@ -17,29 +17,33 @@ def create_graph(model, args):
...
+                tf.compat.v1.global_variables(), name='init')
 
-            tf.train.Saver(tf.global_variables())
+            tf.compat.v1.train.Saver(tf.compat.v1.global_variables())
 
             serialized_graph = session.graph_def.SerializeToString()
             serialized_graph_tensor = tf.convert_to_tensor(serialized_graph)
-            tf.contrib.summary.graph(serialized_graph_tensor, 0, name='graph_write')
-            tf.train.write_graph(
+            tf.contrib.summary.graph(
```
Comment (twuebi) [inline sticker-graph/sticker_graph/write_helper.py:44]: `contrib` will be gone in 2.0, need to check if the method has moved elsewhere.

Comment (danieldk) [reply sticker-graph/sticker_graph/write_helper.py:44]: `tf_upgrade_v2` gives suggestions for replacing `contrib.summary.graph`, but also mentions that it is not a 1:1 conversion. Maybe you could have a look at this sometime after this PR is merged? (I didn't want to touch the summary-writing parts of the graph, since you are the expert ;).)

Comment (twuebi) [reply sticker-graph/sticker_graph/write_helper.py:44]: Should this PR be compatible with 2.0 already?

Comment (danieldk) [reply sticker-graph/sticker_graph/write_helper.py:44]: No, it's just one step towards compatibility (and only the low-hanging fruit). E.g. for CRFs we need to introduce tensorflow-addons as a dependency. There are probably still other incompatibilities besides use of `contrib`.

I also didn't test yet whether the Rust `tensorflow` crate is compatible with TF2.

**Review by twuebi — APPROVED** [review]

**Comment by danieldk** [comment]
Thanks for the reviews! I am doing a small test run on turing-sfb. If everything is fine, I'll merge.

**Comment by danieldk** [comment]
Tested Dutch NER (since it does both RNNs + subword RNN), no regression in accuracy.
